biz.aQute.bndlib/src/aQute/bnd/osgi/repository/AggregateRepository.java

```diff
@@ -0,0 +1,53 @@
+package aQute.bnd.osgi.repository;
+
+import static java.util.Collections.singleton;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+
+import org.osgi.resource.Capability;
+import org.osgi.resource.Requirement;
+import org.osgi.service.repository.Repository;
+
+import aQute.lib.collections.MultiMap;
+
+public class AggregateRepository implements Repository {
+
+	private final Repository repositories[];
+
+	public AggregateRepository(Collection< ? extends Repository> repositories) {
+		this(repositories.toArray(new Repository[repositories.size()]));
+	}
+
+	public AggregateRepository(Repository... repositories) {
+		this.repositories = repositories;
+	}
+
+	@SuppressWarnings({
+			"unchecked", "rawtypes"
+	})
+	@Override
+	public Map<Requirement,Collection<Capability>> findProviders(Collection< ? extends Requirement> requirements) {
+		MultiMap<Requirement,Capability> result = new MultiMap<>();
+
+		for (Repository repository : repositories) {
+			Map<Requirement,Collection<Capability>> capabilities = repository.findProviders(requirements);
+			result.addAll(capabilities);
+		}
+
+		return (Map) result;
```

Comment:
This screams for a method on MultiMap to coerce it from MultiMap<S,T> into Map<S,Collection<T>> rather than you having to make a horrid cast via raw.